lib/ansible/plugins/inventory/packet.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+    name: packet
+    plugin_type: inventory
+    author:
+      - Maxime Guyot (@Miouge1)
+    short_description: Ansible dynamic inventory plugin for Packet.
+    version_added: "2.9"
+    requirements:
+        - python >= 2.7
+        - packet
+    description:
+        - Reads inventories from the Packet API.
+        - Uses a YAML configuration file that ends with packet.(yml|yaml).
+        - The inventory groups are built from plan, tags and facility.
+    options:
+        plugin:
+            description: marks this as an instance of the 'packet' plugin
+            required: true
+            choices: ['packet']
+        auth_token:
+            description: The Packet API auth token
+            required: true
+            env:
+                - name: PACKET_AUTH_TOKEN
+        project_id:
+          description: Project ID
+          required: true
+'''
+
+EXAMPLES = r'''
+# Minimal example. `PACKET_AUTH_TOKEN` is exposed in environment.
+plugin: packet
+
+# Example with regions, types, groups and access token
+plugin: packet
+auth_token: foobar
+regions:
+  - eu-west
+types:
+  - g5-standard-2
+'''
+
+import os
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.module_utils.six import string_types
+from ansible.plugins.inventory import BaseInventoryPlugin
+
+
+try:
+    from packet import Manager
+except ImportError:
+    raise AnsibleError('the Packet dynamic inventory plugin requires packet-python.')
```

Comment:
plugins should not error on import, capture the fact that it is missing the library and exit on 'parse'

Reply:
OK. I've updated this in the latest commit.